Remove manual bindings for arange

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81379
* __->__ #81380
* #81378
* #82241

The functional variant of one of the `arange` overloads has a schema mismatch with the out variant. The functional one has `Scalar step`, but the corresponding out variant has `Scalar step=1`. This isn't allowed, so it had to be special-cased in the python codegen and manually bound. This adds the default `step` value to the functional overload and removes the special-casing.

peterbell10 committed:
Remove manual bindings for arange

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Rebase on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"




[ghstack-poisoned]

ngimel reviewed (approved):
Cool! I don't foresee bc concerns with this one, as all previous invocations should be valid.

peterbell10 committed:
Update on "Remove manual bindings for arange"


The functional variant of one of the `arange` overloads has a schema mismatch with the out variant. The functional one has `Scalar step`, but the corresponding out variant has `Scalar step=1`. This isn't allowed, so it had to be special-cased in the python codegen and manually bound. This adds the default `step` value to the functional overload and removes the special-casing.


[ghstack-poisoned]

peterbell10 committed:
Update on "Remove manual bindings for arange"


The functional variant of one of the `arange` overloads has a schema mismatch with the out variant. The functional one has `Scalar step`, but the corresponding out variant has `Scalar step=1`. This isn't allowed, so it had to be special-cased in the python codegen and manually bound. This adds the default `step` value to the functional overload and removes the special-casing.


[ghstack-poisoned]

peterbell10 committed:
Update xla pin on "Remove manual bindings for arange"


The functional variant of one of the `arange` overloads has a schema mismatch with the out variant. The functional one has `Scalar step`, but the corresponding out variant has `Scalar step=1`. This isn't allowed, so it had to be special-cased in the python codegen and manually bound. This adds the default `step` value to the functional overload and removes the special-casing.


[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge -l
